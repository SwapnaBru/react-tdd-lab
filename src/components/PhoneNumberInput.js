import React from 'react';

export default class PhoneNumberInput extends React.Component {
    constructor() {
        super()

        this.onBlur = this.onBlur.bind(this)
    }

    onBlur() {
        if (this.props.onBlur) {
            this.props.onBlur.apply(this, arguments)
        }
    }

    render() {
        return (
            <div>
                <input type="text" value={this.props.value} onBlur={this.onBlur} />
            </div>
        )
    }
}
