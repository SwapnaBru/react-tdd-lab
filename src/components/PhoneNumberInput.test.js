import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import PhoneNumberInput from './PhoneNumberInput'

describe('PhoneNumberInput', () => {
    describe('no props', () => {
        it('renders without crashing', () => {
            const wrapper = shallow(<PhoneNumberInput />)
            expect(wrapper).toHaveLength(1)
        })

        it('renders correctly', () => {
            const tree = renderer.create(
                <PhoneNumberInput />
            ).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })

    describe('has value', () => {
        it('renders input field with value', () => {
            const wrapper = shallow(<PhoneNumberInput value={"react"} />)
            expect(wrapper.find("input").props().value).toEqual('react')
        })
    })

    describe('has validator', () => {
        it('calls the validator function on blur', () => {
            const validator = jest.fn()
            const wrapper = shallow(<PhoneNumberInput onBlur={validator} />)

            wrapper.find('input').simulate('blur');

            expect(validator.mock.calls.length).toBe(1)
        })
    })
})
