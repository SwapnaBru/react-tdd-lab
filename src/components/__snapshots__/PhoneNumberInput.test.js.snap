// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneNumberInput no props renders correctly 1`] = `
<div>
  <input
    onBlur={[Function]}
    type="text"
    value={undefined}
  />
</div>
`;
